scanner/src/main/kotlin/experimental/ScannerWrapper.kt

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2021 HERE Europe B.V.
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.scanner.experimental
+
+import java.io.File
+
+import org.ossreviewtoolkit.model.KnownProvenance
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScanSummary
+import org.ossreviewtoolkit.model.ScannerDetails
+import org.ossreviewtoolkit.model.config.ScannerConfiguration
+import org.ossreviewtoolkit.scanner.ScanResultsStorage
+import org.ossreviewtoolkit.scanner.ScannerCriteria
+
+/**
+ * An interface to wrap source code scanners.
+ */
+sealed interface ScannerWrapper {
+    /**
+     * The name of the scanner.
+     */
+    val name: String
```

Comment:
`name` is already part of `ScannerDetails`. Is there really a need to duplicate this here?

Reply:
It's not strongly required, but `details` depends on `getVersion()` which could be slow, e.g. if it depends on calling the CLI tool or a network call in the case of FossId.

Reply:
So, are there use-cases where you want to get the scanner name (fast), and don't need the other scanner details?

Reply:
I'm not sure, the name is used in several log statements, for example, but on the other hand the `details` should be built only once. Shall I remove `name` for now and if we find any performance issues we re-introduce it?